src/jsvm-ext.cpp

```diff
@@ -10,71 +10,52 @@
  */
 
 #include "jsvm-ext.h"
-
 #include <m_printf.h>
 
-#define jerryx_port_handler_print_char(A) m_putc(A)
+using namespace Jerryscript;
 
-jerry_value_t alertFunction(const jerry_call_info_t *call_info_p,
-							const jerry_value_t args_p[], const jerry_length_t args_count)
+namespace JS
 {
-	jerry_value_t str_val = jerry_value_to_string(args_p[0]);
-
-	jerry_size_t req_sz = jerry_get_string_size(str_val);
-	jerry_char_t str_buf_p[req_sz + 1];
-
-	jerry_string_to_char_buffer(str_val, str_buf_p, req_sz);
-	str_buf_p[req_sz] = '\0';
-
-	m_printf("%s", (char*)str_buf_p);
-	jerry_release_value(str_val);
-
-	return jerry_create_boolean(true);
+Value alertFunction(const CallInfo& callInfo, Value& str)
+{
+	String s = str.toString();
+	m_nputs(s.c_str(), s.length());
+	m_putc('\n');
+	return true;
 }
 
-jerry_value_t printFunction(const jerry_call_info_t *call_info_p,
-							const jerry_value_t args_p[], const jerry_length_t args_count)
+Value printFunction(const CallInfo&, Value args[], unsigned argCount)
 {
-	static const char* null_str = "\\u0000";
```

Comment:
I'm not sure why the original implementation of this bothered display NULL differently, can't think of any particular reason (other than debugging I guess). It complicated the code so just removed it.

Reply:
Ok.